public/src/components/ProgressBar.tsx

```diff
@@ -93,3 +94,39 @@ const ProgressBar = ({ courseID, submission, type }: ProgressBarProps) => {
 }
 
 export default ProgressBar
+
+// defaultProgressBar is a function that returns a progress bar for a lab/assignment with no submissions
+// It is used in the AssignmentCard component
+export const defaultProgressBar = (submissions: Submission[], assignment: Assignment) => {
```

Comment:
It seems like there are some opportunities for code reuse between this one and the `ProgressBar` type above. Couldn't some of this be extracted out and reused by passing appropriate parameters?